### ISSUE TITLE
x/tools/gopls: handle line directives to .y files

### ISSUE THREAD

**Issue by renthraysk** [golang/go#34433]
### What version of Go are you using (`go version`)?
go version go1.13 linux/amd64

And gopls...
golang.org/x/tools/gopls v0.1.7
    golang.org/x/tools/gopls@v0.1.7 h1:YwKf8t9h69++qCtVmc2q6fVuetFXmmu9LKoPMYLZid4=

### Does this issue reproduce with the latest release?
Yes

### What did you do?
Open https://play.golang.org/p/-QfVqDKxsFI in VSCode.

### What did you expect to see?
No crashing.

### What did you see instead?
"The gopls server crashed 5 times in the last 3 minutes. The server will not be restarted."

**Comment by stamblerre** [comment]
Thanks for reporting this. Looks like we aren't correctly handling code with line directives. This will require some work, but we can at least prevent the crash from occurring.

**Comment by heschik** [comment]
The work I'm doing for #35720 is a good start on this one, but we're not tracking .y files so I don't think it'll actually fix it. We'd have to be able to get a ColumnMapper for an arbitrary non-Go file.

**Comment by pjweinb** [comment]
This no longer crashes, instead writing an error message

Failed to compute document links: must supply Converter for file ".../foof.go" containing lines from ".../expr.y"

My personal opinion is that it will be almost impossible to get //line processing right, although maybe a few special cases could be done (after 1.0 is released).

**Comment by stamblerre** [comment]
I agree that handling line directives is not a critical priority for v1.0. Moving this out of the 1.0 milestone.

**Comment by pjweinb** [comment]
resolved: safetoken  safely ignores virtual //line directives
